### ISSUE TITLE
Items that were previously unscrobbled do not appear in the "Biggest climbers" lists

### ISSUE THREAD

**Issue by TheOptimisticFactory** [felhag/lastfm-stats-web#47]
Hi,

After stumbing on this project today, and being amazed by the stats offered, I noticed something rather odd in the `biggest climbers` part of the tracks.

# Current results

- For my profile, it currently showcases these 3 songs
![image](https://github.com/felhag/lastfm-stats-web/assets/2607260/63dc5bdf-9203-4a6b-8dce-18f1a0bca6c1)

- If we take a deeper look at the n°1 result, we can see the following metrics: `from unplayed, then 1 scrobble the next month, and then 101 scrobbles the next`
![image](https://github.com/felhag/lastfm-stats-web/assets/2607260/ce46b128-5d52-445a-8776-94380b610539)

- And even weirder the 10th entry on the `biggest climbers` is a song having as low as `11 scrobbles in a single month`

# Expectations

- I would have expected this song to be in the 1st spot:  `from unplayed to 326 scrobbles in a single month`
![image](https://github.com/felhag/lastfm-stats-web/assets/2607260/505c85ff-86bf-49d7-af8f-624fe62fe878)

# Question

- What dictates that a given track ranks higher than others in the `biggest climbers`?

**Comment by TheOptimisticFactory** [comment]
After digging, I got the answer: 
- to be registered in the `biggest climbers/fallers` an artists/album/track has to be scrobbled 2 consecutive months, and have the largest rank difference in terms of scrobbles.

Meaning: 
- an unplayed artist/album/track won't be registered on its first month of being scrobbled. 

I think having a newly discovered song (scrobbles:0) should be counted in the `biggest climbers/fallers`

**Comment by felhag** [comment]
Thanks for the clear issue :). You're right, climbers/fallers are only counted from the second month they appear. It doesn't really have a ranking if it's not scrobbled I figured... And (maybe, I'm not sure how it is ordered for artists/albums/tracks with the same amount of scrobbles) the list will be filled with climbers from 0 to 1 scrobbles which would be kinda boring.

I don't think the months have to be consecutively though. If so that is clearly a bug.

**Comment by TheOptimisticFactory** [comment]
> _the list will be filled with climbers from 0 to 1 scrobbles which would be kinda boring_

That is a fair point.

Maybe some middleground could be found, where a threshold of scrobbles has to be reached for a newly discovered to be considered in the climber (ideally based on the average scrobble count of the profile).

This would allow an unplayed song to be registered in the biggest climbers assuming it got played enought on the month of its discovery.

In my case having one song going from unplayed to literal rank n°1 in a single month would be certainly be above said threshold. The current implementation occults it entirely, as the rank remains `1` between the discovery month and the following.

**Comment by felhag** [comment]
Yeah, I agree, a threshold would be a nice solution. A similar threshold is used for some other lists (for example golden oldies, average scrobble per track). But it's not dynamically determined, it's 50 for artists, 30 for albums and 10 for tracks.

**Comment by felhag** [comment]
I added the threshold (50 for artists, 30 for albums and 10 for tracks), but didn't include the unscrobbled month into the biggest climbers/fallers lists (yet?). Do you think this solves your issue? If not you can reopen the issue it if you like.

**Comment by TheOptimisticFactory** [comment]
Thanks @felhag, that sounds promising!

I will try this out and report if the results are matching expectations.
Is https://lastfmstats.com up-to-date with these changes?

**Comment by felhag** [comment]
@TheOptimisticFactory, yes, it's up to date!

**Comment by TheOptimisticFactory** [comment]
> I added the threshold (50 for artists, 30 for albums and 10 for tracks), but didn't include the unscrobbled month into the biggest climbers/fallers lists (yet?). Do you think this solves your issue? If not you can reopen the issue it if you like.

I imported my profile on https://lastfmstats.com just now, and I must say that I did not notice much differences compared to the previous results. 

To clarify:

- The current algorith is valid, and works as intended: it displays items that indeed climbed the ranking rapidly, **`UNDER THE ASSUMPTION`** that item got scrobbled at least once in the past.

- But I would still expect are **`unscrobbled`** items to be showcased in the biggest climbers/fallers assuming the climb is over a given threshold. it could be an even bigger threshold than for already-scobbled items, to make sure they do not pollute the results.

I am re-opening the issue in hope to hopefully see **`unscrobbled`** items to get the recognition they deserve :)

---

Side-note: @felhag your tool is awesome by the way.

**Comment by felhag** [comment]
I reopened it, I'll have another look at it.
